layouts/partials/blocks/candidate-results.html

```diff
@@ -0,0 +1,80 @@
+{{ with .block }}
+  {{ $data := index site.Data.candidates2024 }}
+
+  <script>
+    function candidates() {
+
+      let urlParams = new URL(window.location).searchParams.entries();
+      let candidateData = {{$data}}
+
+      function paramsToObject() {
```

Comment:
I think this is the same as [`Object.fromEntries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries), so you could write this instead:

```js
let candidateScores = Object.fromEntries(new URL(window.location).searchParams.entries())
```

Test it out and LMK if it works.